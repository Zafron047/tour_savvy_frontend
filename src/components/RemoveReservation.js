import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { removeReservation } from '../redux/reservations/reservationSlice';
import '../stylesheets/delReservation.css';
import '../stylesheets/common.css';

function RemoveReservation() {
  const dispatch = useDispatch();
  const { reservations, isLoading } = useSelector(
    (store) => store.reservations,
  );
  const handleRemove = async (e, id) => {
    e.preventDefault();
    dispatch(removeReservation(id));
  };

  const user = JSON.parse(localStorage.getItem('user'));
  if (!user) {
    return <div>Please log in to delete a reservation.</div>;
  }

  if (isLoading) {
    return (
      <>
        <div>
          <h2>Loading...</h2>
        </div>
      </>
    );
  }
  return (
    <div className="delReservation">
      <div className="background" />
      <div className="main-delReservation">
        <h2 className="cancel-heading">Cancel a reservation</h2>
        {reservations.map((reservation) => (
          <div className="cancel-card" key={reservation.id}>
            <div className="cancel-details">
              <h2>
                Location:&nbsp;
                {reservation.city_name}
              </h2>
              <h3>
                Date:&nbsp;
                {reservation.reservation_date}
              </h3>
              <h3>
                Package:&nbsp;
                {reservation.package_type}
              </h3>
            </div>
            <button
              type="button"
              className="form-btns cancel-btn"
              onClick={(e) => handleRemove(e, reservation.id)}
            >
              Cancel
            </button>
          </div>
        ))}
      </div>
    </div>
  );
}

export default RemoveReservation;
