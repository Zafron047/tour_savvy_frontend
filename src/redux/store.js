import { configureStore } from '@reduxjs/toolkit';
import packagesSlice from './packages/packagesSlice';
import reservationsReducer from './reservations/reservationSlice';
import userSlice from './authentication/userSlice';

const loadStateFromLocalStorage = () => {
  try {
    const serializedState = localStorage.getItem('reduxState');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

const loadUserFromLocalStorage = () => {
  try {
    const serializedUser = localStorage.getItem('user');
    if (serializedUser === null) {
      return undefined;
    }
    return JSON.parse(serializedUser);
  } catch (err) {
    return undefined;
  }
};

const store = configureStore({
  reducer: {
    packages: packagesSlice,
    reservations: reservationsReducer,
    current_user: userSlice,
  },
  preloadedState: {
    current_user: loadUserFromLocalStorage(),
    ...loadStateFromLocalStorage(),
  },
  reservations: reservationsReducer,
});

export default store;
