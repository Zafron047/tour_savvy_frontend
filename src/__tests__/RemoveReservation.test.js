import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import RemoveReservation from '../components/RemoveReservation';

test('Should render RemoveReservation component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <RemoveReservation />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
