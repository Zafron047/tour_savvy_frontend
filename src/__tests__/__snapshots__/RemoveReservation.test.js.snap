// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render RemoveReservation component correctly 1`] = `
<div>
  <div>
    Please log in to delete a reservation.
  </div>
</div>
`;
